### PR TITLE
Update RSC15b for 0.9

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

File: Spec/RestClient.swift [hunk 1]
```diff
@@ -646,6 +646,39 @@ class RestClient: QuickSpec {
 
                     expect(NSRegularExpression.match(capturedURLs[1], pattern: "//[f-j].ably-realtime.com")).to(beTrue())
                 }
+
+                // RSC15b
+                it("applies when ClientOptions#fallbackHostsUseDefault is true") {
+                    let options = ARTClientOptions(key: "xxxx:xxxx")
```
Comment (mattheworiordan) [inline Spec/RestClient.swift]: `fallbackHostsUseDefault` being true does nothing unless a custom host / environment is being used.  You need to specify a custom env / host as well as `fallbackHostsUseDefault`

Comment (ricardopereira) [inline Spec/RestClient.swift]: Done.

**Comment by ricardopereira** [comment]
@mattheworiordan Please have a new look. I changed some code because tests were not passing.

File: Source/ARTClientOptions.m [hunk 2]
```diff
@@ -126,11 +126,19 @@ - (BOOL)isBasicAuth {
 }
 
 - (BOOL)hasCustomRestHost {
-    return _restHost != nil;
+    return _restHost || _environment;
```
Comment (tcard) [inline Source/ARTClientOptions.m]: Maybe also `![_restHost equals:[ARTDefault restHost]]`?

**Comment by mattheworiordan** [comment]
LGTM

File: Source/ARTClientOptions.m (hunk 2 above)
Comment (ricardopereira) [inline Source/ARTClientOptions.m]: Absolutely! Thanks.

Comment (ricardopereira) [inline Source/ARTClientOptions.m]: Done 6ba9147.

**Pull request merged**